tests/test_middleware_interface.py

```diff
@@ -910,17 +910,17 @@ def test_filter_datasets(self):
         # Case where src is empty now covered in test_filter_datasets_nosrc.
         for src, existing in itertools.product(combinations, [set()] + combinations):
             diff = src - existing
-            src_butler = unittest.mock.Mock(
-                **{"registry.queryDatasets.return_value.expanded.return_value": src})
-            existing_butler = unittest.mock.Mock(**{"registry.queryDatasets.return_value": existing})
+            src_butler = unittest.mock.Mock(**{"query_datasets.return_value": src})
+            existing_butler = unittest.mock.Mock(**{"query_datasets.return_value": existing})
 
             with self.subTest(src=sorted(ref.dataId["detector"] for ref in src),
                               existing=sorted(ref.dataId["detector"] for ref in existing)):
                 result = set(_filter_datasets(src_butler, existing_butler,
                                               ["bias"], instrument="LSSTComCamSim"))
-                src_butler.registry.queryDatasets.assert_called_once_with("bias", instrument="LSSTComCamSim")
-                existing_butler.registry.queryDatasets.assert_called_once_with("bias",
-                                                                               instrument="LSSTComCamSim")
+                src_butler.query_datasets.assert_called_once_with(
+                    "bias", instrument="LSSTComCamSim", explain=False)
+                existing_butler.query_datasets.assert_called_once_with(
+                    "bias", instrument="LSSTComCamSim", explain=False)
```

Comment:
The `explain=False` is an implementation detail. Unfortunately, I don't know how to write a test without it other than iterating over `Mock.call_args`.